### PR TITLE
Fix pre-commit hook failing because of black

### PR DESCRIPTION
**Pull request by Phlair**
## What
See [this issue](https://github.com/airbytehq/airbyte/issues/11493) and [this PR](https://github.com/airbytehq/airbyte/pull/11494).

I've just updated the black version in the pre-commit hook to match the upgrade done there.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ repos:
       - id: licenseheaders
         args: ["--tmpl=LICENSE_SHORT", "--ext=py", "-f"]
   - repo: https://github.com/ambv/black
```
Comment (alafanechere) [inline .pre-commit-config.yaml:10]: nit:
```suggestion
  - repo: https://github.com/psf/black
```

Looks like there's a redirection to a newest repo. (maybe the username changed)

**Review by alafanechere — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Phlair) [reply .pre-commit-config.yaml:10]: oh good spot

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -7,8 +7,8 @@ repos:
     hooks:
       - id: licenseheaders
         args: ["--tmpl=LICENSE_SHORT", "--ext=py", "-f"]
-  - repo: https://github.com/ambv/black
-    rev: 22.1.0
+  - repo: https://github.com/psf/black
```
Comment (supertopher) [inline .pre-commit-config.yaml:10]: I have no context and am just curious: why did we switch maintainers?  A quick look at both GitHub pages didn't give me a clear answer

Comment (Phlair) [reply .pre-commit-config.yaml:10]: `https://github.com/ambv/black` is now redirecting to `https://github.com/psf/black` so it's not an actual maintainer switch, arguably unnecessary but just for clarity.

Comment (supertopher) [reply .pre-commit-config.yaml:10]: Cool.  Cool cool cool. thanks for the context

**Pull request merged**